core/gesture_handler.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The class that handlers gestures (drags, clicks, etc) across
+ * all of Blockly, including multiple workspaces.
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+/**
+ * @name Blockly.GestureHandler
+ * @namespace
+ */
+goog.provide('Blockly.GestureHandler');
+
+goog.require('Blockly.Gesture');
```

Comment:
Does this already exist somewhere? I think you're just breaking up changes into pieces which is cool, but I want to make sure I'm not missing a piece myself :)